lib/Service/RoomService.php

```diff
@@ -551,4 +551,13 @@ public function setMessageExpiration(Room $room, int $seconds): void {
 
 		$this->dispatcher->dispatch(Room::EVENT_AFTER_SET_MESSAGE_EXPIRATION, $event);
 	}
+
+	public function setShowHistory(Room $room, int $showHistory): void {
```

Comment:
**Required:** 

Needs to throw if the value is not 0 or 1 for now.
Should dispatch evens like with all other room attributes as well